### PR TITLE
feat: Make it easier to discover how to add documentation to swagger.…

### LINKED ISSUE #4620: Make it easier to discover how to add documentation to swagger.json
# Summary

Add a contact field to the top of the swagger.json file which links to a page that explains how to update the file (and potentially how to file bugs/etc.)

# Motivation

For #4616, I went through a lot of trouble trying to figure out how to properly update the `swagger.json` file.

# Proposal

Add a `contact` field to the `info` element of `swagger.json`. The `url` field could link to anything, it could be to a file in the repo, or a wiki, or something else.
The `email` field is mandatory, and I don't have a particular opinion about where it should point.

### PR DESCRIPTION
**Pull request by jsoref**
Closes #4620

Signed-off-by: Josh Soref <jsoref@users.noreply.github.com>

Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [x] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [ ] I've signed the CLA and my build is green ([troubleshooting builds](https://argoproj.github.io/argo-cd/developer-guide/ci/)).

### CODE REVIEW

**Comment by jannfis** [comment]
I like that :+1:

However, since the URL this PR references does not exist (yet), and I'm unsure whether we should (re-)activate the GH Wiki. I think we can discuss this on Thursday's contributor meeting, along with other developer-related documentation which does not necessarily has to live in Git (because it's updated just too often, possibly).

**Comment by jsoref** [comment]
Yeah. I have no opinion about where any of this should live. Nor about the `name` field.

If the wiki is abandoned, then I'd encourage the project to delete and disable it. I believe there's also a website repository.

**Comment by pasha-codefresh** [comment]
@jsoref are you going to finalize this PR?

**Comment by pasha-codefresh** [comment]
Maybe lets bring it on Thursday contribution meeting?

**Comment by jsoref** [comment]
The information for that is apparently a bit hard to follow.

https://github.com/argoproj/argo-cd/blob/3091bc5b6f3d389376feaa999a63fc4986ae72a9/docs/developer-guide/code-contributions.md#regular-contributor-meeting

It looks like it's 11:15 am Eastern which is precisely when my team has its daily.

I can see about attending, but it's a problematic ask.

And if I do it, I'd want to discuss both of these tickets.

**Comment by pasha-codefresh** [comment]
okay, but it would be nice to take some decision around these items, because they stay opened from 2020y, and we need to understand if we want close them or continue work. Thank you

**Comment by jsoref** [comment]
Yeah. I'll ask today about skipping our daily.

**Comment by todaywasawesome** [comment]
Moving to draft until there's more attention here.

**Comment by jsoref** [comment]
I've added this to my calendar. The fact that the meeting time collides with my standup is unfortunate...

**Comment by jsoref** [comment]
We're going to add a page w/:

* where are the proto files
* where are the go struct type files
* what's the make target

**Comment by pasha-codefresh** [comment]
@jsoref any plans with this PR, should we close it ?

**Comment by pasha-codefresh** [comment]
@jsoref , there has been no activity on this PR for a long time. I am closing it. Feel free to reopen once you decide to continue working on it.
